src/dmd/lexer.d

```diff
@@ -202,6 +202,129 @@ unittest
     }
 }
 
+
+/**************
+ * Pool allocator for Token objects, used by the Lexer.
+ * Tokens are never freed so Token pointers will never change
+ */
+private struct TokenPool{
+private:
+    enum PoolSize = 1024; //number of tokens per pool
+    enum DefaultPoolListSize = 64; //number of pools to allocate by default
+    Token[][] pools;
+    size_t poolsAllocated = 0;
+
+    nothrow:
+
+    ///Token pool looks like a big array
+    public ref Token getByIndex(size_t index){
+        if (index >= PoolSize*poolsAllocated)
+        {
+            allocatePools(index + 1); //need space for index 0
+        }
+        return pools[index / PoolSize][index % PoolSize];
+
+    }
+
+    private void allocatePools(size_t tokensNeeded){
+        const poolsNeeded = (tokensNeeded + PoolSize - 1)/PoolSize;
+        assert(poolsNeeded == poolsAllocated + 1); //don't alloc more than 1 pool at a time
+        if (poolsNeeded > pools.length)
+        {
+            const newLength = pools.length > 0 ? 2*pools.length : DefaultPoolListSize;
```

Comment:
The new size should be a factor of 1.5, not 2. Compute it by multiplying by 3 then dividing by 2. Round up.